### DISCUSSION TITLE
Transformation command for complex geometry

### DISCUSSION THREAD

**Discussion by kmmm70** [neperfepx/neper#230] [Q&A] [answered]
Hi All,

I wonder how to use -transform to create a complex geometry model. I tried

`neper -T -n 200 -transform "cut(sphere(24,3,6,5),sphere(10,3,6,5))" -domain "cube(48,6,12)"`  ===> it's OK.

`neper -T -n 100 -transform "planecut(48,1,1,1)" -domain "cube(48,6,12)"` ===> It's OK.

When I tried to combine them together with

`neper -T -loadtess n200-id1.tess -transform "planecut(48,1,1,0)"` ===> It's failed showing error. 

How to use different transformation together to create complex geometry model?

Thanks in advance.

**Comment by rquey** [comment, marked as answer]
I'm not sure what's happening here, but you can use `cut(hspace)` instead of `planecut` (they are almost equivalent), and it will work:

```
$ neper -T -n 200 -transform "cut(sphere(24,3,6,5),sphere(10,3,6,5),hspace(48,1,1,0))" -domain "cube(48,6,12)"
```

**Reply by kmmm70** [reply]
Romain,

It's amazing. Much appreciate for your help. I found this picture from Not real engineering on youtube. Do you think I can manipulate to get that model using the combination of -transform "cut()" ?

Thanks a lot.
<img width="243" alt="2021-05-26_15-15-05" src="https://user-images.githubusercontent.com/57537961/119626359-509de000-be35-11eb-8039-aa7650d61a1e.png">

**Reply by kmmm70** [reply]
By the way, How can I replace the circles with square or rectangular shape?

**Reply by rquey** [reply]
This image was the image of the homepage before version 4. It is an (elongated) cube cut by 2 cylinders (for the holes) and 2 other cylinders for the gauge section. The circles cannot be replaced by squares or rectangles.

**Reply by kmmm70** [reply]
Romain,

I've got it. Thanks for your fast reply. Will you consider making it cut by the other shapes in the future?

**Reply by rquey** [reply]
Not in the near future. This is a limitation of the current algorithm.
